zk/proofs/pol/src/lottery.rs

```diff
@@ -1,56 +1,36 @@
 use std::{
     ops::{Deref as _, Div as _},
-    str::FromStr as _,
     sync::LazyLock,
 };
 
 use num_bigint::BigUint;
-use num_traits::CheckedSub as _;
+use num_traits::{CheckedSub as _, Num as _};
 
-#[expect(clippy::too_long_first_doc_paragraph, reason = "we need it")]
-/// :warning: There may be dragons
-/// The [following constants](https://www.notion.so/nomos-tech/Proof-of-Leadership-Specification-21c261aa09df819ba5b6d95d0fe3066d?source=copy_link#256261aa09df800fbc88e5aae5ea7e06)
-/// Use the str representation instead of the hex representation from the
-/// specification because of types signed/unsigned representations. Constants
-/// where computed in python which uses a signed representation of the integer.
-/// `BigInt` type (signed), parsed the hex values correctly. But `BigUint` does
-/// not for the motives just described. Using the `str` representation allows
-/// use to choose the unsigned type.
+/// From [Proof of Leadership spec](https://www.notion.so/nomos-tech/Proof-of-Leadership-Specification-21c261aa09df819ba5b6d95d0fe3066d?source=copy_link#256261aa09df800fbc88e5aae5ea7e06)
 pub static P: LazyLock<BigUint> = LazyLock::new(|| {
-    BigUint::from_str(
-        "21888242871839275222246405745257275088548364400416034343698204186575808495617",
+    BigUint::from_str_radix(
+        "30644e72e131a029b85045b68181585d2833e84879b9709143e1f593f0000001",
+        16,
```

Comment:
Does this work?

Reply:
[Yes?](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=8afb6dc653e4541c8811f3c3447dd443)
I could add tests as well if necessary but it's a bit of a strange to put there, I don't see why this shouldn't work 🤔